geps/gep-91/index.md

```diff
@@ -1,22 +1,164 @@
+* Also introduce a `ObjectReference` structure that can be used to specify `caCertificateRefs` references.
+
+#### GO
+
+```go
+// ObjectReference identifies an API object including its namespace.
+//
+// The API object must be valid in the cluster; the Group and Kind must
+// be registered in the cluster for this reference to be valid.
+//
+// References to objects with invalid Group and Kind are not valid, and must
+// be rejected by the implementation, with appropriate Conditions set
+// on the containing object.
+type ObjectReference struct {
+	// Group is the group of the referent. For example, "gateway.networking.k8s.io".
+	// When unspecified or empty string, core API group is inferred.
+	Group Group `json:"group"`
+
+	// Kind is kind of the referent. For example "ConfigMap" or "Service".
+	Kind Kind `json:"kind"`
+
+	// Name is the name of the referent.
+	Name ObjectName `json:"name"`
+
+	// Namespace is the namespace of the referenced object. When unspecified, the local
+	// namespace is inferred.
+	//
+	// Note that when a namespace different than the local namespace is specified,
+	// a ReferenceGrant object is required in the referent namespace to allow that
+	// namespace's owner to accept the reference. See the ReferenceGrant
+	// documentation for details.
+	//
+	// Support: Core
+	//
+	// +optional
+	Namespace *Namespace `json:"namespace,omitempty"`
+}
+
+type GatewayTLSConfig struct {
+    ......
+    // FrontendValidation holds configuration information for validating the frontend (client).
+    // Setting this field will require clients to send a client certificate
+    // required for validation during the TLS handshake. In browsers this may result in a dialog appearing 
+    // that requests a user to specify the client certificate.
+    // The maximum depth of a certificate chain accepted in verification is Implementation specific.
+    FrontendValidation *FrontendTLSValidation `json:"frontendValidation,omitempty"`
+}
+
+// FrontendTLSValidation holds configuration information that can be used to validate
+// the frontend initiating the TLS connection
+type FrontendTLSValidationContext struct {
+    // CACertificateRefs contains one or more references to
+    // Kubernetes objects that contain TLS certificates of
+    // the Certificate Authorities that can be used
+    // as a trust anchor to validate the certificates presented by the client.
+    //
+    // A single CA certificate reference to a Kubernetes ConfigMap
+    // has "Core" support.
+    // Implementations MAY choose to support attaching multiple CA certificates to
+    // a Listener, but this behavior is implementation-specific.
+    //
+    // Support: Core - An optional single reference to a single Kubernetes Secret
+    // and to a single Kubernetes ConfigMap with the CA certificate in a key named `ca.crt`.
```

Comment:
Nit: I'm not sure we really want to consider Secrets core support, that would mean we'd need to have conformance tests for both ConfigMaps and Secrets when we really just want people to use ConfigMaps and eventually ClusterTrustBundles.

Reply:
ah yes, thanks for finding this, fixed, was a left over, which I forgot to clean up